inginious/frontend/plugins/grader_generator/static/js/grader_generator.js

```diff
@@ -53,13 +53,33 @@ function insertDataContent(templateElement, test_data) {
         }
     );
 
+    templateElement.find(`#${baseId}_input_file`).attr("value", test_data["input_file"]);
+    templateElement.find(`#${baseId}_output_file`).attr("value", test_data["output_file"]);
+
     templateElement.find(`#${baseId}_diff_shown`).prop('checked', test_data["diff_shown"]);
 }
 
+function removeTestByFilename(filename) {
+    const testId = getIdByFilename(filename);
+    if (testId !== undefined) {
+        studio_remove_test_case(testId);
+    }
+}
+
+function getIdByFilename(fileName) {
+    const inputId = $(`input[value='${fileName}']`).attr("id");
+    if (inputId !== undefined) {
+        return getIdNum(inputId);
+    }
+}
+
 
 function getIdNum(id) {
     const baseIdPrefixLen = "grader_test_cases_".length;
-    return id.substr(baseIdPrefixLen);
+    const idString = id.substr(baseIdPrefixLen);
+    /*parseInt returns a number from a string if the number is at the beginning of the string.
+    It doesn't matter if the string has more characters*/
+    return parseInt(idString, 10);
```

Comment:
Is it possible that the whole does not contain any number? Does this raise an exception?

Reply:
It will be strange, but if it happen, the function retuns `NaN`

Reply:
So do you mean that this use case may happen or will it never happen? (when the string does not contain a number) In case that is possible and the function returns `NaN`, does not this brakes something?

Reply:
It will be strange because the ids have a pattern and the number is achieved taking into account that pattern, so a string without a number at the beginning will never happen. But if that happens, the worst case is that no test is removed, and first I will think there are problems with the ids.